strawberry/experimental/pydantic/object_type.py

```diff
@@ -48,19 +67,38 @@ def replace_pydantic_types(type_: Any):
 
         return new_type
 
+    if issubclass(type_, Enum):
+        new_type = strawberry.enum(type_)
+        return new_type
+
     if issubclass(type_, BaseModel):
         if hasattr(type_, "_strawberry_type"):
             return type_._strawberry_type
         else:
-            raise UnregisteredTypeException(type_)
+            if register_nested is not None:
+                # register nested class
+                type_name = register_nested.format_type(type_.__name__)
+                strawberry.experimental.pydantic.type(
+                    model=type_, all_fields=True, register_nested=register_nested
+                )(create_type(type_name, (), {}))
+
+                assert hasattr(
+                    type_, "_strawberry_type"
+                ), f"Failed to create _strawberry_type for {type_.__name__}. Please report this bug"
+
+                return type_._strawberry_type
+            else:
+                raise UnregisteredTypeException(type_)
```

Comment:
Logic here can be simplified and # of returns reduced if we reverse the `if/else`, plus I think it reads a bit better:

```python
if not hasattr(type_, "_strawberry_type"):
    ...
return type_._strawberry_type
```